eras/conway/impl/src/Cardano/Ledger/Conway/Rules/Epoch.hs

```diff
@@ -256,26 +256,47 @@ epochTransition = do
         , reCurrentEpoch = eNo
         }
     govStateToSeq = Seq.fromList . Map.toList
-    ratSig = RatifySignal . govStateToSeq . unGovActionsState $ cgGovActionsState govSt
-  rs@RatifyState {rsFuture} <-
-    trans @(EraRule "RATIFY" era) $ TRC (ratEnv, cgRatifyState govSt, ratSig)
-  let es'' =
-        epochState'
-          { esAccountState = acnt''
-          , esLState = (esLState epochState') {lsUTxOState = utxoSt'''}
-          }
-          & prevPParamsEpochStateL .~ pp
-          & curPParamsEpochStateL .~ pp
-      -- TODO can we be more efficient?
-      newGov = GovActionsState . Map.fromList . toList $ rsFuture
-      esGovernanceL = esLStateL . lsUTxOStateL . utxosGovStateL
-      esDonationL :: Lens' (EpochState era) Coin
-      esDonationL = esLStateL . lsUTxOStateL . utxosDonationL
-      donations = es'' ^. esDonationL
+    ratSig = RatifySignal . govStateToSeq . prevGovActionsState $ cgGovActionsState govSt
+  RatifyState {rsRemoved, rsEnactState} <-
+    trans @(EraRule "RATIFY" era) $
+      TRC
+        ( ratEnv
```

Comment:
I suspect that we need to supply `reCurrentEpoch = eNo - 1` to ratify instead of a current epochNo, otherwise all DReps and GovActions will expire one epoch sooner.

Moreover, once you rebase on master, you'll see that RatifyEnv now also includes DRep state, which means we need to snapshot that as well, otherwise DReps can deregister during the epoch. Same applies to ComitteeState, we'll have to snapshot that, otherwise votes for resigned CC members will not count.

We need to discuss it more with @WhatisRT